### PR TITLE
[Bugfix][Parser] Hide incomplete Muse Glimmer ATEM markers

### LINKED ISSUE #55395: [Bug]: Muse Glimmer tool parser leaks incomplete ATEM markers
### Your current environment

Current `vllm-project/vllm` main, Muse Glimmer reasoning and tool parsers enabled, streaming chat completions.

### Describe the bug

If generation ends after a proper prefix of an ATEM control marker—for example, `<atem:function_calls` without the final `>`—the legacy Muse Glimmer tool parser can emit that fragment as ordinary visible content. Split markers can also briefly be committed before the next chunk completes the tag.

ATEM framing is protocol control data, not user-visible content. An incomplete final marker should be dropped, and a split marker should be withheld until it either completes or is proven to be ordinary text.

### Reproduction

For each ATEM marker, split it at every character boundary and call `MuseGlimmerToolParser.extract_tool_calls_streaming()` twice. Concatenated visible content should contain only the preceding user answer. Also terminate after every proper prefix and verify that only the preceding answer is emitted.

Affected markers include:

- `<atem:function_calls>` / `</atem:function_calls>`
- `<atem:invoke` / `</atem:invoke>`
- `<atem:parameter` / `</atem:parameter>`

### Expected behavior

No complete or partial ATEM control marker reaches visible response content. Complete invokes continue through normal tool-call parsing; incomplete markers do not fabricate calls.

### Related work

PR #54585 migrates the reasoning half to the Streaming Parser Engine and appears to cover its `<atem:` framing. Its description explicitly leaves the legacy ATEM tool parser unchanged. This issue tracks only that complementary tool-parser behavior to avoid duplicating #54585.

### PR DESCRIPTION
**Pull request by lming2001**
## Purpose

Prevent the legacy Muse Glimmer tool parser from exposing complete or partial ATEM protocol markers as ordinary streamed content.

A marker split across chunks is held until it completes. If generation terminates after a proper marker prefix, the incomplete control fragment is dropped rather than surfaced or interpreted as a fabricated tool call.

Fixes #55395.

This is not duplicating an existing PR: #54585 migrates only the reasoning parser and explicitly leaves the legacy ATEM tool parser unchanged. This draft is the complementary tool-parser fix and depends on coordinating with that migration.

## Tests

```bash
python -m pytest -q tests/tool_use/test_muse_glimmer.py
pre-commit run --files \
  tests/tool_use/test_muse_glimmer.py \
  vllm/tool_parsers/muse_glimmer_tool_parser.py
```

Results:

- Muse Glimmer parser suite: **33 passed**.
- All applicable pre-commit hooks passed, including Ruff, mypy, SPDX, import checks, and configuration validation.
- Added split-at-every-boundary and final-truncation coverage for all function-call, invoke, and parameter ATEM markers.

No model weights or logits are changed. This is a protocol-layer output fix; model-quality evaluation is not applicable. The original real-traffic reproduction ended after a partial `<atem:function_calls` opener and leaked it into visible content.

## AI assistance

Agentcloud was used to help port the previously validated fix and run tests. This PR remains draft until the human submitter reviews every changed line and can explain and defend the change end to end.

---

<details>
<summary>Essential Elements of an Effective PR Description Checklist</summary>

- [x] The purpose of the PR and linked issue.
- [x] The test plan and results.
- [x] Why this does not duplicate an existing PR.
- [ ] Human submitter has reviewed every changed line.

</details>

### CODE REVIEW

**Comment by lming2001** [comment]
Full 987-record TPU validation is complete.

- Direct stack (#3527 + #3528 + this PR): performance is baseline-consistent (c8 QPS +0.3%, E2E P50 +0.8%, P95 +0.4%), but three c8 responses still end in partial `<atem:function_calls` because this PR intentionally does not modify the reasoning parser.
- Dependency-complete stack with #54585: zero incomplete ATEM content leaks across c4+c8, but #54585 currently reduces QPS by 21.7% at c4 and 15.2% at c8, with TTFT P50 increasing from 324ms→1,988ms and 543ms→3,441ms.
- Both stacks completed 979/987 at each cell; the same eight prompts returned expected context-limit 400s. No empty outputs occurred.

I am keeping this PR draft. Its tool-parser change passes 33 legacy Muse tests, the dependency-complete parser stack passes 58 focused tests, and the full-corpus regression evidence is posted on #54585. This PR should not be marked ready until the human submitter reviews every changed line and the dependency's frontend regression is resolved.

**Comment by lming2001** [comment]
Closing this draft as superseded by #55419, which addresses the same incomplete ATEM marker leakage with broader edge-case coverage. The separate reasoning-parser migration remains tracked in #54585.
